### PR TITLE
Lon 1250

### PR DESCRIPTION
**Pull request by vguzmanp**

### CODE REVIEW

File: VAS.UI.Gtk2/UI/Common/TreeViewBase.cs [hunk 1]
```diff
@@ -256,15 +256,24 @@ protected virtual void AddSubViewModel (IViewModel subViewModel, TreeIter parent
...
-			} else {
-				dictionaryStore [subViewModel].ForEach (x => store.Remove (ref x));
-				dictionaryStore.Remove (subViewModel);
+			if (subViewModel is INestedViewModel)
+			{
+				RemoveAllNestedSubViewModels(subViewModel);
+			}
+			else
+			{
+				TreeIter iter = dictionaryStore[subViewModel].First();
```
Comment (ylatuya) [inline VAS.UI.Gtk2/UI/Common/TreeViewBase.cs]: What happens for example with the players treeview in which an event can be in several players ?

Comment (vguzmanp) [reply VAS.UI.Gtk2/UI/Common/TreeViewBase.cs]: That was exactly the case that made this crash. The event was in each player's timeline, so each of them was calling remove, and the second one was crashing.
Now the first one will remove the first one, and the next calls will end up removing all of them.

File: VAS.UI.Gtk2/UI/Common/TreeViewBase.cs [hunk 2]
```diff
@@ -263,8 +263,13 @@ protected virtual void RemoveSubViewModel (IViewModel subViewModel)
 			if (subViewModel is INestedViewModel) {
 				RemoveAllNestedSubViewModels (subViewModel);
 			} else {
-				dictionaryStore [subViewModel].ForEach (x => store.Remove (ref x));
-				dictionaryStore.Remove (subViewModel);
+				TreeIter iter = dictionaryStore [subViewModel].First ();
```
Comment (ylatuya) [inline VAS.UI.Gtk2/UI/Common/TreeViewBase.cs]: What happens in the players treeview where the same timelineeventvm is un several playervm? you will be removing just the first one, aren't you?

Comment (vguzmanp) [reply VAS.UI.Gtk2/UI/Common/TreeViewBase.cs]: (copypasted) That was exactly the case that made this crash. The event was in each player's timeline, so each of them was calling remove, and the second one was crashing.
Now the first one will remove the first one, and the next calls will end up removing all of them.

**Pull request merged**